### PR TITLE
Add a special page to generate graphs

### PR DESCRIPTION
**Pull request by cicalese**

### CODE REVIEW

File: src/EntryPoints/SpecialNetwork.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		$params[] = 'labelMaxLength=' . strval( $labelMaxLength );
+
+		/**
+		 * @psalm-suppress PossiblyNullReference
+		 */
+		$html = ( new NetworkFunction( $config ) )->handleParserFunctionCall(
+			$output,
+			$output->getTitle()->getFullText(),
+			$params
+		);
```
Comment (JeroenDeDauw) [inline src/EntryPoints/SpecialNetwork.php:69]: It is nice how you reused things here. The design is meant to do this on the UseCase level though. So both the parser function and the special page would create the UseCase and call it. The special page would not use the parser function. This means some of the wiring code will be duplicated, but as you noticed, not everything is the same. In cases where the wiring is more complex, this duplication could be reduced by extracting it into a factory. Probably not warranted here yet.

File: src/EntryPoints/SpecialNetwork.php [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+			];
+
+			$htmlForm =
+				HTMLForm::factory( 'ooui', $formDescriptor, $this->getContext() );
+
+			$htmlForm->setMethod( 'get' );
+
+			$htmlForm->prepareForm()->displayForm( false );
+		}
+	}
```
Comment (JeroenDeDauw) [inline src/EntryPoints/SpecialNetwork.php:128]: This function is a doing a bunch of different stuff and really could use some splitting.

File: src/EntryPoints/SpecialNetwork.php (hunk 1 above)
Comment (cicalese) [reply src/EntryPoints/SpecialNetwork.php]: Agreed. I did some refactoring to NetworkPresenter to facilitate reusing at the UseCase level. Let me know what you think. I am still not completely happy with the interaction with Extension.

File: src/EntryPoints/SpecialNetwork.php (hunk 2 above)
Comment (cicalese) [reply src/EntryPoints/SpecialNetwork.php:194]: Agreed. I did some major refactoring that resulted in more code but has better separation of concerns.

File: src/EntryPoints/SpecialNetwork.php [hunk 3]
```diff
@@ -22,108 +27,169 @@ public function __construct() {
...
 		}
+	}
 
-		$class = $request->getText( 'class' );
-		if ( $class !== '' ) {
-			$params[] = 'class=' . $class;
+	/**
+	 * @param WebRequest $request
+	 * @param NetworkConfig $config
+	 * @return string[] $params
```
Comment (JeroenDeDauw) [inline src/EntryPoints/SpecialNetwork.php]: Not an issue - personally I'd omit the parameters since now the info is duplicated

**Comment by krabina** [comment]
Wow this is great! My first thoughts:
- it would be important to be able to exclude not only individual pages but also namespaces
- css classes and graph options fields should be collapsed first and only visible after a click somehwere (they look confusinig for a regular user)
This is a real research interface, I love it!

**Comment by cicalese** [comment]
> * it would be important to be able to exclude not only individual pages but also namespaces
 
See #29

**Pull request merged**